File: Top_Interview_Questions/validSudoku.go
func isValidSudoku(board [][]byte) bool {
    seen := map[string]bool{}
    for row := 0; row < 9; row++ {
        for col := 0; col < 9; col++ {
            num := string(board[row][col])
            if num != "." {
                rowKey := fmt.Sprint(num, "row", row)
                if _, exists := seen[rowKey]; exists {
                    return false
                } else {
                    seen[rowKey] = true
                }
                colKey := fmt.Sprint(num, "col", col)
                if _, exists := seen[colKey]; exists {
                    return false
                } else {
                    seen[colKey] = true
                }
                boxKey := fmt.Sprint(num, "box", row/3, "-", col/3)
                if _, exists := seen[boxKey]; exists {
                    return false
                } else {
                    seen[boxKey] = true
                }
            }
        }
    }
    return true
}

File: NeetCode_Roadmap/valid_sudoku.py
class Solution:
    def isValidSudoku(self, board: List[List[str]]) -> bool:
        seen = set()
        for row in range(len(board)):
            for col in range(len(board[0])):
                if board[row][col] == ".":
                    continue
                # row
                key = f"r{row}:{board[row][col]}"
                if key in seen:
                    return False
                seen.add(key)
                # column
                key = f"c{col}:{board[row][col]}"
                if key in seen:
                    return False
                seen.add(key)

                # subbox
                key = f"b{row // 3}-{col // 3}:{board[row][col]}"
                if key in seen:
                    return False
                seen.add(key)

        return True